src/app.go

```diff
@@ -20,57 +25,124 @@ type Config struct {
 	HTTP     *server.Config
 	Logger   *zap.Config
 	Postgres *client.Config
-	OIDC     *jwt.Config
-	APIKey   *apikey.Config
+	OIDC     *jose.Config
+	APIKey   *csv.Config
 	TLS      *tls.Config
-	Manifest *manifests.Config
+	Manifest *manifestreader.Config
 }
 
-func New(cfg *Config, logger log.Logger) (*app.App, error) {
-	// Create app
-	a := app.New(&app.Config{HTTP: cfg.HTTP}, logger.WithComponent("app"))
-
-	// Register Service Configuration
-	err := a.RegisterServiceConfig(&authapp.Config{Manifest: cfg.Manifest, OIDC: cfg.OIDC, TLS: cfg.TLS, APIKey: cfg.APIKey})
+func New(ctx context.Context, cfg *Config, logger log.Logger) (*app.App, error) {
+	// Infra layer
+	pgClient, err := client.New(cfg.Postgres)
 	if err != nil {
 		return nil, err
 	}
 
-	err = a.RegisterServiceConfig(&storesapp.Config{Postgres: cfg.Postgres, Manifest: cfg.Manifest})
+	jwtValidator, err := getJWTValidator(cfg.OIDC, logger)
+	if err != nil {
+		return nil, err
+	}
+	apikeyClaims, err := getAPIKeys(ctx, cfg.APIKey, logger)
 	if err != nil {
 		return nil, err
 	}
 
-	err = a.RegisterServiceConfig(&nodesapp.Config{Manifest: cfg.Manifest})
+	rootCAs, err := getRootCAs(ctx, cfg.TLS, logger)
 	if err != nil {
 		return nil, err
 	}
 
-	err = a.RegisterServiceConfig(&aliasapp.Config{Postgres: cfg.Postgres})
+	manifests, err := getManifests(ctx, cfg.Manifest)
 	if err != nil {
 		return nil, err
 	}
 
+	a := app.New(&app.Config{HTTP: cfg.HTTP}, logger.WithComponent("app"))
+
 	// Register Services
-	err = authapp.RegisterService(a, logger.WithComponent("auth"))
+	authService, err := authapp.RegisterService(ctx, a, logger.WithComponent("auth"), manifests, jwtValidator, apikeyClaims, rootCAs)
 	if err != nil {
 		return nil, err
 	}
 
-	err = storesapp.RegisterService(a, logger.WithComponent("stores"))
+	aliasService, err := aliasapp.RegisterService(a, logger.WithComponent("aliases"), pgClient)
 	if err != nil {
 		return nil, err
 	}
 
-	err = aliasapp.RegisterService(a, logger.WithComponent("aliases"))
+	storesService, err := storesapp.RegisterService(ctx, a, logger.WithComponent("stores"), pgClient, manifests, authService)
 	if err != nil {
 		return nil, err
 	}
 
-	err = nodesapp.RegisterService(a, logger.WithComponent("nodes"))
+	_, err = nodesapp.RegisterService(ctx, a, logger.WithComponent("nodes"), manifests, authService, storesService, aliasService)
 	if err != nil {
 		return nil, err
 	}
 
 	return a, nil
 }
+
+func getAPIKeys(ctx context.Context, cfg *csv.Config, logger log.Logger) (map[string]*authtypes.UserClaims, error) {
```

Comment:
Auth service constructors(getAPIKEys, getJWTValidator and getRootCAs) must be located under Auth package for better encapsulation of responsibilities instead of having them in `app.go`.